packages/react-native-editor/__device-tests__/gutenberg-editor-writing-flow-@canary.test.js

```diff
@@ -0,0 +1,348 @@
+		await editorPage.typeTextToTextBlock( captionInput, '\n' );
+
+		// Expect to have an empty Paragraph block and the keyboard visible
+		let paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph,
+			2
+		);
+		expect( paragraphBlockElement ).toBeTruthy();
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( true );
+		await editorPage.typeTextToTextBlock(
+			paragraphBlockElement,
+			backspace
+		);
+
+		// When deleting the Paragraph block, the keyboard should be hidden and
+		// the image block should be focused.
+		await editorPage.driver.waitUntil( async function () {
+			return ! ( await editorPage.driver.isKeyboardShown() );
+		} );
+		expect( await editorPage.isImageBlockSelected() ).toBe( true );
+
+		// Adding a new Paragraph block
+		await editorPage.addNewBlock( blockNames.paragraph );
+		paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph,
+			2
+		);
+
+		// It should be focused and the keyboard should be visible
+		expect( paragraphBlockElement ).toBeTruthy();
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( true );
+
+		await editorPage.typeTextToTextBlock(
+			paragraphBlockElement,
+			testData.shortText
+		);
+
+		const imageBlockElement = await editorPage.getBlockAtPosition(
+			blockNames.image
+		);
+		await imageBlockElement.click();
+
+		await editorPage.driver.waitUntil( async function () {
+			return ! ( await editorPage.driver.isKeyboardShown() );
+		} );
+		expect( await editorPage.isImageBlockSelected() ).toBe( true );
+	} );
+
+	it( 'should manually dismiss the keyboard', async () => {
+		await editorPage.initializeEditor( { initialTitle: '' } );
+
+		const defaultBlockAppenderElement =
+			await editorPage.getDefaultBlockAppenderElement();
+		await defaultBlockAppenderElement.click();
+
+		const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph
+		);
+		expect( paragraphBlockElement ).toBeTruthy();
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( true );
+
+		await editorPage.dismissKeyboard();
+
+		// Checks that no block is selected by looking for the block menu actions button
+		expect( await editorPage.isBlockActionsMenuButtonDisplayed() ).toBe(
+			false
+		);
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( false );
+	} );
+
+	it( 'should dismiss the keyboard and show it back when opening modals', async () => {
+		await editorPage.initializeEditor( { initialTitle: '' } );
+
+		const defaultBlockAppenderElement =
+			await editorPage.getDefaultBlockAppenderElement();
+		await defaultBlockAppenderElement.click();
+
+		await editorPage.openBlockSettings();
+		await editorPage.driver.waitUntil( async function () {
+			return ! ( await editorPage.driver.isKeyboardShown() );
+		} );
+
+		await editorPage.dismissBottomSheet();
+
+		await editorPage.driver.waitUntil( async function () {
+			return await editorPage.driver.isKeyboardShown();
+		} );
```

Comment:
Nitpick:
```suggestion
		await editorPage.driver.waitUntil( editorPage.driver.isKeyboardShown );
```

Reply:
Nice suggestion! Addressed in https://github.com/WordPress/gutenberg/pull/57460/commits/7e6fa2d0646b078f3b2a2ba143b9d3d157caa0d6